File: Tipo_de_dato/TipoElemento.cs
namespace Estructuras_de_datos.Tipo_de_dato
{
    public class TipoElemento
    {
        private int _clave;
        private object? _valor;



        public TipoElemento(int clave)
        {
            _clave = clave;
            _valor = null;
        }

        public TipoElemento(int clave, object valor)
        {
            _clave = clave;
            _valor = valor;
        }

        //Getters
        public int Clave { get { return _clave; } }
        public object? Valor { get { return _valor; } }


    }
}

File: Referencias/tipo_elemento.c
#include <stdio.h>
#include <stdlib.h>
#include "tipo_elemento.h"

//Escribo las funciones
TipoElemento te_crear(int clave)
{
    TipoElemento te = (TipoElemento) malloc(sizeof(struct TipoElementoRep));
    te->clave = clave;
    te->valor = NULL;
    return te;
}

TipoElemento te_crearConValor(int clave, void* valor)
{
    TipoElemento te = (TipoElemento) malloc(sizeof(struct TipoElementoRep));
    te->clave = clave;
    te->valor = valor;
    return te;
}
